frontend/src/modules/search/components/SearchBox.js

```diff
@@ -65,73 +65,86 @@ export class SearchBoxBase extends React.Component<InternalProps, State> {
     constructor(props: InternalProps) {
         super(props);
 
+        this.state = {
+            search : '',
+            statuses: {},
+            extras: {},
+            tags: {},
+            timeRange: null,
+            authors: {},
+        };
+        
+        this.searchInput = React.createRef();
+    }
+
+    componentDidMount() {
+        // $FLOW_IGNORE (errors that I don't understand, no help from the Web)
+        document.addEventListener('keydown', this.handleShortcuts);
+        this.initializeFilters();
+    }
+
+    componentDidUpdate(prevProps: InternalProps) {
+        // Clear search field when navigating to a new file
+        if (
+            this.props.parameters.search === null &&
+            prevProps.parameters.search !== null
+        ) {
+            this.setState({
+                search: '',
+            });
+        }
+    }
+
+    componentWillUnmount() {
+        // $FLOW_IGNORE (errors that I don't understand, no help from the Web)
+        document.removeEventListener('keydown', this.handleShortcuts);
+    }
+
+    initializeFilters = () => {
+        // Reset filters to state provided by the url parameters
```

Comment:
You should define a `props` constant at the beginning of the function and then you can keep the diff much shorter by keeping `props...` instead of `this.props...`.